js/src/evaluation/evaluate_comparative.ts

```diff
@@ -332,12 +333,37 @@ export async function evaluateComparative(
     return result;
   }
 
+  const tracedEvaluators = options.evaluators.map((evaluator) =>
+    traceable(
+      async (
+        runs: Run[],
+        example: Example
+      ): Promise<ComparisonEvaluationResultRow> => {
+        const evaluatorRun = getCurrentRunTree();
+        const result = await evaluator(runs, example);
+
+        // sanitise the payload before sending to LangSmith
+        evaluatorRun.inputs = { runs: runs, example: example };
+        evaluatorRun.outputs = result;
+
+        return {
+          ...result,
+          source_run_id: result.source_run_id ?? evaluatorRun.id,
+        };
+      },
+      {
+        project_name: "evaluators",
```

Comment:
Do we need to do anything if they have their tracing vars turned off

Reply:
that should be already dealt by `tracingIsEnabled` inside `traceable`